editoast/src/views/path/projection.rs

```diff
@@ -407,111 +407,98 @@ mod tests {
-        ];
-        let projection = PathProjection::new(&path);
-
-        let track_ranges = vec![
-            TrackRange::new("A", 0, 100, Direction::StartToStop),
-            TrackRange::new("B", 0, 200, Direction::StartToStop),
-            TrackRange::new("X", 0, 100, Direction::StartToStop),
-            TrackRange::new("C", 150, 200, Direction::StopToStart),
-            TrackRange::new("Z", 0, 100, Direction::StartToStop),
-            TrackRange::new("E", 30, 100, Direction::StartToStop),
-        ];
-
-        let boundaries = projection.get_intersections(&track_ranges);
-        let expected: Vec<Intersection> = vec![
-            Intersection::from((50, 300)),
-            Intersection::from((400, 450)),
-            Intersection::from((550, 620)),
-        ];
-
-        assert_eq!(boundaries, expected);
-    }
-
-    #[test]
-    fn get_boundaries_case_4() {
-        let path = vec![TrackRange::new("A", 0, 100, Direction::StartToStop)];
-        let projection = PathProjection::new(&path);
-
-        let track_ranges = vec![TrackRange::new("B", 0, 100, Direction::StartToStop)];
-
-        let boundaries = projection.get_intersections(&track_ranges);
-
-        let expected: Vec<Intersection> = vec![];
-        assert_eq!(boundaries, expected);
-    }
-
-    #[test]
-    fn get_boundaries_case_5() {
-        let path = vec![TrackRange::new("A", 0, 100, Direction::StartToStop)];
+    #[rstest]
+    // One track on the path
+    #[case::one_path_different_track(&["A+0-100"], &["B+0-100"], &[])]
+    #[case::one_path_no_overlap(&["A+0-100"], &["A+100-200"], &[])]
+    #[case::one_path_one_simple_intersection(&["A+120-140"], &["A+100-200"], &[(20, 40)])]
+    #[case::one_path_one_simple_intersection_reverse_on_track_ranges(&["A+140-120"], &["A+100-200"], &[(20, 40)])]
+    #[case::two_path_merged(&["A+180-200", "B+100-120"], &["A+100-200", "B+100-200"], &[(80, 120)])]
+    #[case::two_path_not_merged(&["A+180-220", "B+80-120"], &["A+100-200", "B+100-200"], &[(80, 120)])]
+    #[case::two_path_merged_with_extra_bounds(&["A+180-220", "B+80-120"], &["A+100-200", "B+100-200"], &[(80, 120)])]
+    #[case::three_path_with_hole(&["A+150-200", "C+100-150"], &["A+100-200", "B+100-200", "C+100-200"], &[(50, 100), (200, 250)])]
+    // Complex paths with complex track ranges
+    #[case::complex_path_one_intersection(
+        &["A+50-100", "B+200-0", "C+0-300", "D+250-120"],
+        &["A+0-100", "B+200-0", "C+0-300", "D+250-0", "E+0-100"],
+        &[(50, 730)]
+    )]
+    #[case::complex_path_two_intersections(
+        &["A+50-100", "B+200-0", "C+0-300", "D+250-0", "E+100-25"],
+        &["X+0-100", "B+0-200", "C+200-150", "E+30-100", "Z+0-100"],
+        &[(100, 350), (350, 420)]
+    )]
+    #[case::complex_path_three_intersections(
+        &["A+50-100", "B+200-0", "C+0-300", "D+250-0", "E+100-25"],
+        &["A+0-100", "B+0-200", "X+0-100", "C+200-150", "Z+0-100", "E+30-100"],
+        &[(50, 300), (400, 450), (550, 620)]
+    )]
```

Comment:
Could you please add a comment in the test code to explain how it works, especially the business logic part? It would help a lot in understanding it.

Reply:
I've made a few comments in multiple places in [3750ea8](https://github.com/OpenRailAssociation/osrd/pull/10137/commits/3750ea87f7a87178d9504f2b3e0cd30b0a112cec), what do you think?

Reply:
Looks good, thanks for the updates!